android/src/main/java/com/bitmovin/player/reactnative/RNPlayerView.kt

```diff
@@ -44,6 +104,73 @@ class RNPlayerView(context: ReactApplicationContext) : LinearLayout(context) {
         if (playerView.parent != this) {
             (playerView.parent as ViewGroup?)?.removeView(playerView)
             addView(playerView)
+            startBubblingEvents()
+        }
+        pictureInPictureHandler?.let {
+            if (it.isPictureInPictureAvailable) {
```

Comment:
I think this if check might be a bug.
What if PiP is disabled when `addPlayerView` is called, but later it is enabled 🤔 (and of course always supported by the system)

Reply:
Yeah, your point makes sense 🤔. If PiP is later re-enabled, then this method won't be called again unless the React `PlayerView` is also re-rendered. In other words, `addPlayerView` is only called when the react component is mounting. Maybe exporting the PiP option as a [`@ReactProp`](https://reactnative.dev/docs/native-components-android#3-expose-view-property-setters-using-reactprop-or-reactpropgroup-annotation) can do the trick, since it's called anytime the prop changes. I'll do some testing to see if any of the assumptions are right and I'll get back with the results later. Thanks 🙂.

Reply:
So you’re right about setting `isPictureInPictureEnabled` late not changing the player’s state, but that’s not because of the `if` statement. Actually, that happens because we don’t connect the player’s config with the underlying component state on React inside `usePlayer`.

We use `useRef` to maintain the same player instance alive across renderings, but changing its state (like config, for example) won’t cause the UI to re-render. We would need `useState` or `useReducer` for this, but enabling this behavior in the current state of the library isn’t so easy from what I could see. I think we should move this work to a different PR.

Trying to change `useRef` with `useState` to keep track of the player’s config has led to some errors, and I still don’t know what’s causing it 😕.

Reply:
But we could still remove the if check?

Reply:
Yeah, sure 👍